engines/python/setup/djl_python/rolling_batch/trtllm_rolling_batch.py

```diff
@@ -63,17 +63,33 @@ def inference(self, input_data, parameters):
         # step 0: register new active requests
         for request in new_requests:
             param = self.translate_triton_params(request.parameters)
+            output_len = param["request_output_len"]
             response = self.model.generate(request.input_text, **param)
-            self.request_cache[request.id] = response
+            self.request_cache[request.id] = {
+                "response": response,
+                "out_length": output_len,
+                "cumulative_logprob": 0
+            }
 
         # step 1: loop the active requests to send result
         for request in self.active_requests:
-            trt_req = self.request_cache[request.id]
-            output_text, complete = trt_req.fetch()
-            request.set_next_token(output_text, self.output_formatter,
-                                   complete)
-            if complete:
+            trt_resp = self.request_cache[request.id]["response"]
+            generation = trt_resp.fetch()
+            log_prob = generation.cum_logprob - self.request_cache[
+                request.id]["cumulative_logprob"]
+            self.request_cache[
+                request.id]["cumulative_logprob"] = generation.cum_logprob
+            token = Token(generation.token_id, generation.token_text, log_prob,
+                          None)
+            if generation.finished:
+                finish_reason = "eos_token" if generation.seq_length < self.request_cache[
+                    request.id]["out_length"] else "length"
```

Comment:
nit: i think we should call finished_reason as `max_length` instead of `length` if max_length is reached to make it more readable

Reply:
This is the openAI default name, TGI also used a similar name